src/index/Index.h

```diff
@@ -281,4 +281,9 @@ class Index {
   // requires including the rather expensive `IndexImpl.h` header
   IndexImpl& getImpl() { return *pimpl_; }
   [[nodiscard]] const IndexImpl& getImpl() const { return *pimpl_; }
+
+  /*
+  @brief Print the detailed documentation of the options for the index builder.
+  */
+  std::string getConfigurationDocForIndexBuilder();
```

Comment:
I think this should return also two strings: the JSON and the detailed configuration separately, then we can configure the printing in `IndexBuilderMain` (I will think of something).

Reply:
Shouldn't the configuration of the printing happen inside `IndexImpl::getConfigurationDocForIndexBuilder`? Separation of concerns and all that.